### ISSUE TITLE
Problem installing manual setup for development

### ISSUE THREAD

**Issue by bigredfrog** [inkstitch/inkstitch#1293]
following instructions at 

https://inkstitch.org/developers/inkstitch/manual-setup/

On a farm fresh ubuntu 21.04 VM

Fails on installing the requirements

big_red_frog@inkstitch:~/dev/inkstitch$ pip install -r requirements.txt 
ERROR: Directory './pyembroidery' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.

Any hints? Don't believe I have missed any prior steps

Cloned inkstich
Cloned pyembroidery into the same directory
Installed pyembroidery
changed directory to institch, tried the requiremetns call and it falls apart...

**Comment by kaalleen** [comment]
Ah yes, we need to update this page. For now you can just comment pyembroidery out in the requirements file, since you have it already installed. Sorry for your trouble.

**Comment by bigredfrog** [comment]
Thanks for this, having removed the first line reference to pyembroidery, things proceeded a lot further, but then went red line.

Building wheel for wxPython (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-8pvtqfg0/wxpython_65ee7b588f38484da02a81848088e6d5/setup.py'"'"'; __file__='"'"'/tmp/pip-install-8pvtqfg0/wxpython_65ee7b588f38484da02a81848088e6d5/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-nq32yunw
       cwd: /tmp/pip-install-8pvtqfg0/wxpython_65ee7b588f38484da02a81848088e6d5/
  Complete output (144 lines):

Assuming this is related to the guidance in the manual install around wxpython could someone help me with more explicit guidance? I am on ubuntu 21.04

Instructions are 

Info: You might need to remove wxPython and install a platform specific package:
⚫ Debian uses python-wxgtk3.0
⚫ Ubuntu 16.04: pip install -U -f https://extras.wxpython.org/wxPython4/extras/linux/gtk3/ubuntu-16.04 wxPython

But,order of operation here is not clear to me, now this has failed, do I need to do a sudo apt remove wxPython?
Then some variant of the install line? Then would I rerun dependancies, or is this an ok end state.

Thanks for you help

**Comment by bigredfrog** [comment]
Grasping at straws here, but do I need an older Ubuntu release? everything I see if referencing python2
I will try a 16.04 vm and see if it behaves better..

**Comment by bigredfrog** [comment]
Back to 21.04, think I am swimming too deep in the past on 16.04Had to run
sudo apt-get install build-essential libgtk-3-dev

To get me past the wxPython complaint

**Comment by bigredfrog** [comment]
OK, so the libgtk-3-dev manual install then allowed the requirements.txt to run to completion.

Next onto Electron and yarn

When trying to run as per

cd electron
yarn install

I get 

big_red_frog@inkstitch:~/dev/inkstitch/electron$ yarn install
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'

Also tried 

big_red_frog@inkstitch:~/dev/inkstitch/electron$ yarn
00h00m00s 0/0: : ERROR: There are no scenarios; must have at least one.

But not sure I have yarn installed correctly.

Will keep scratching, but clues are appreciated

**Comment by bigredfrog** [comment]
yeah, it was yarn from cmdtest

Couldn't get npm install on yarm to launch, so not sure what is wrong there

So tried the following and I have the next step spinning

sudo apt remove cmdtest
sudo apt remove yarn
curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
sudo apt-get update
sudo apt-get install yarn -y

Now the yarn install failed with

big_red_frog@inkstitch:~/dev/inkstitch/electron$ yarn install
yarn install v1.22.5
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] Resolving packages...
[2/4] Fetching packages...
error An unexpected error occurred: "Commit hash required".
info If you think this is a bug, please open a bug report with the information provided in "/home/big_red_frog/dev/inkstitch/electron/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

Clues on a postcard...

**Comment by bigredfrog** [comment]
[yarn-error.log](https://github.com/inkstitch/inkstitch/files/6909444/yarn-error.log)

**Comment by bigredfrog** [comment]
oh wow, moved back to an npm install of yarn, but could not find the command.

Closed, and opened new command window, yarn launches fine, version 1.22.11

Same error though "Commit hash required"

**Comment by bigredfrog** [comment]
I will keep putting my ramblings in here to capture the trail, and once I get to "success" I will summarise at the top.

Deleted the yarn.lock file

Reran yarn install and things are spinning

**Comment by bigredfrog** [comment]
Attempting to run make inx, I get an error, and no inx files

big_red_frog@inkstitch:~/dev/inkstitch$ make inx
bash bin/generate-version-file
# message files will look like this:
#   translations/messages_en_US.po
if ls translations/*.po > /dev/null 2>&1; then \
	for po in translations/*.po; do \
		lang=${po%.*}; \
		lang=${lang#*_}; \
		mkdir -p locales/$lang/LC_MESSAGES/; \
		msgfmt $po -o locales/$lang/LC_MESSAGES/inkstitch.mo; \
	done; \
else \
	mkdir -p locales; \
fi
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
/bin/sh: 6: msgfmt: not found
make: *** [Makefile:41: locales] Error 127

**Comment by bigredfrog** [comment]
Needed to install gettext to pick up msgfmt

sudo apt install gettext

Now struggling with getting the python call into generate-inxfiles to run

Question I assume over which python to install

**Comment by bigredfrog** [comment]
OK, so this error syntax is 

print(xxx, file=yyy)

Is python3 and python2 will error as reported, so now I need to ensure I am only python3 I assume

I thought I read somewhere that inkstitch was python2?...

**Comment by bigredfrog** [comment]
switched default python with

sudo su
update-alternatives --install /usr/bin/python python /usr/bin/python3 1

Then had to 

pip install --upgrade lxml
pip install --upgrade webencodings

I now have the needed .mo file in the locals directory structure.

Getting close!

**Comment by bigredfrog** [comment]
Almost there.

I can see the inkstitch extensions menu in inkscape after setting up the symbolic link

I can load a file I know to be good in normal inkstitch, but on launching for example the simulator view I get

Traceback (most recent call last):
  File "inkstitch.py", line 16, in <module>
    sys.path.remove('/usr/share/inkscape/extensions')
ValueError: list.remove(x): x not in list

So I still have something basic wrong...

**Comment by bigredfrog** [comment]
I can see through trivial modifciation it is my source tree running, when I purposefully break things, my diagnostic is in the dialog, so that is good.

However, it really is falling at the first, and I think my spelunking has bottomed out

What is this python code actually doing as the remove and append seem indenticle?
and it is the remove that fails on me.

if getattr(sys, 'frozen', None) is None:
    # When running in development mode, we want to use the inkex installed by
    # pip install, not the one bundled with Inkscape which is not new enough.
    sys.path.remove('/usr/share/inkscape/extensions')
    sys.path.append('/usr/share/inkscape/extensions')

**Comment by bigredfrog** [comment]
If I bypass this code I then hit the shapely error of no module named shapely

Going to have to turn away from this for a bit now, going snowblind.

Is it that the instructions have rotted, or am I doing something silly, as this seems a lot of dancing above to get here...

**Comment by bigredfrog** [comment]
Confirmed by messing with exit string the extension is attempting to run

sys.version_info(major=3, minor=8, micro=10, releaselevel='final', serial=0)

Where as my python from the terminal is 

big_red_frog@inkstitch:~$ python --version
Python 3.9.5

So there is something to fix there, but I don't know that it is related to my remove / append problem...

**Comment by bigredfrog** [comment]
Ha, is my problem that I installed inkscape as a snap?

 ['/home/big_red_frog/dev/inkstitch', '/home/big_red_frog/.config/inkscape/extensions', '/snap/inkscape/9090/share/inkscape/extensions', '/snap/inkscape/9090/share/inkscape/extensions/inkex/deprecated-simple', '/usr/lib/python38.zip', '/usr/lib/python3.8', '/usr/lib/python3.8/lib-dynload', '/snap/inkscape/9090/lib/python3.8/site-packages']

So '/usr/share/inkscape/extensions'  will not be in that list...

**Comment by bigredfrog** [comment]
Yep, installed from ppa

sudo add-apt-repository universe
sudo add-apt-repository ppa:inkscape.dev/stable
sudo apt-get update
sudo apt install inkscape

Now the next failure is., I am assuming still to do with the version of inkex

Ink/Stitch Version: Manual Install (Linux)

Traceback (most recent call last):
  File "/home/big_red_frog/dev/inkstitch/inkstitch.py", line 59, in <module>
    extension.run(args=remaining_args)
  File "/home/big_red_frog/.local/lib/python3.9/site-packages/inkex/base.py", line 141, in run
    self.save_raw(self.effect())
  File "/home/big_red_frog/dev/inkstitch/lib/extensions/simulator.py", line 17, in effect
    if not self.get_elements():
  File "/home/big_red_frog/dev/inkstitch/lib/extensions/base.py", line 175, in get_elements
    self.elements = nodes_to_elements(self.get_nodes(troubleshoot))
  File "/home/big_red_frog/dev/inkstitch/lib/extensions/base.py", line 172, in get_nodes
    return self.descendants(self.document.getroot(), troubleshoot=troubleshoot)
  File "/home/big_red_frog/dev/inkstitch/lib/extensions/base.py", line 159, in descendants
    nodes.extend(self.descendants(child, selected, troubleshoot))
  File "/home/big_red_frog/dev/inkstitch/lib/extensions/base.py", line 141, in descendants
    element.get_style('display', 'inline') is None):
  File "/home/big_red_frog/dev/inkstitch/lib/elements/element.py", line 171, in get_style
    style = self._get_specified_style().get(style_name, default)
  File "/home/big_red_frog/dev/inkstitch/lib/elements/element.py", line 168, in _get_specified_style
    return self.node.specified_style()
  File "/home/big_red_frog/.local/lib/python3.9/site-packages/inkex/elements/_base.py", line 151, in __getattr__
    raise AttributeError("Can't find attribute {}.{}".format(self.typename, name))
AttributeError: Can't find attribute Layer.specified_style

**Comment by bigredfrog** [comment]
did a pip uninstall inkex to remove the local instance.

Now inkscape just ties up when I call the simulation view against a known good svg

Trying the troubleshoot pass on the same file gives me

Ink/Stitch experienced an unexpected error.

If you'd like to help, please file an issue at https://github.com/inkstitch/inkstitch/issues and include the entire error description below:

Ink/Stitch Version: Manual Install (Linux)
Traceback (most recent call last):

  File "/home/big_red_frog/dev/inkstitch/inkstitch.py", line 59, in <module>
    extension.run(args=remaining_args)
  File "/home/big_red_frog/dev/inkstitch/src/inkscape-core-extensions/inkex/base.py", line 147, in run
    self.save_raw(self.effect())
  File "/home/big_red_frog/dev/inkstitch/lib/extensions/troubleshoot.py", line 23, in effect
    self.create_troubleshoot_layer()
  File "/home/big_red_frog/dev/inkstitch/lib/extensions/troubleshoot.py", line 106, in create_troubleshoot_layer
    add_layer_commands(layer, ["ignore_layer"])
  File "/home/big_red_frog/dev/inkstitch/lib/commands.py", line 391, in add_layer_commands
    svg = layer.root()
TypeError: 'SvgDocumentElement' object is not callable

So at this point I think I am at the mercy of "those that know"

Any help appreciated...

**Comment by bigredfrog** [comment]
One more data point before I crawl off to find beer.

If I select some of the most complex geometry, the params dialog and the associated preview dialog works!

But Troubleshoot and simualtion do not.

**Comment by bigredfrog** [comment]
random question, could my deletion of yarn.lock of been a bad thing on building electron?

It simply would not build prior with 

"Commit hash required"

**Comment by bigredfrog** [comment]
I am probably doing this wrong, but on a hunch I moved to the v2.0.0 tag on origin/main

I think that worked, but I may have it wrong.

Anyway, troubleshoot now completes without error and tells me I have no issues in my sample file, which is what I would hope

Unfortunately simulation still dies on me just hangs up.

**Comment by bigredfrog** [comment]
Also after moving to the v2.0.0 tag, exporting to .pes is sane again, I don't get the unexpected layer, just exactly what I expect from the layers that are active. So it would appear there are regressions in head.

**Comment by bigredfrog** [comment]
Final note for today.

I reran yarn install in electron just in case there were branch changes there, but no change.

Also I cannot run the pdf generation, I just get these warnings after some runtime

/home/big_red_frog/.local/lib/python3.9/site-packages/colormath/color_diff.py:67: DeprecationWarning: np.asscalar(a) is deprecated since NumPy v1.16, use a.item() instead

  return numpy.asscalar(delta_e)

/home/big_red_frog/dev/inkstitch/lib/extensions/print_pdf.py:95: UserWarning: The 'environ['werkzeug.server.shutdown']' function is deprecated and will be removed in Werkzeug 2.1.

  request.environ.get('werkzeug.server.shutdown')()

I sincerely want to be able to help, but need to be bump started a little for baseline

**Comment by kaalleen** [comment]
Wow, you have been quite active here. Sorry that we didn't make it easy for you to get your manual install running. The website needs some attention.

Since Ink/Stitch v2.0 we use Python 3, no Python 2 involved anymore.

The wxpython comment on the website is also a bit outdated and should be removed. I am currently not using Ubuntu, so I am not really sure which version of wxpython needs to be installed there. But for my system I can simply use the pip installed version of wxpython without any trouble.

Your simulator issues are caused by the nodejs version you are using (16.x I assume). We don't use the most recent version, due to an issue with fibers. I just created a branch, to make it possible to use the most recent nodejs version: kaalleen/update-nodejs. Try this one.

The pip version of inkex hasn't been updated for a while, but we needed their update to be able to recognize the correct styling of certain objects. Using the git inkex version has been causing issues for you. Since you were switchng Ink/Stitch versions, it could be possible, that you have inkex installed via pip. Verify with `pip show inkex`. If it is installed run `pip uninstall inkex`.
Btw. removing and appending the very same path will set it to the end of the list, so it will only be used if the other options haven't been found. `AttributeError: Can't find attribute Layer.specified_style` indicates that something went wrong with that step and it doesn't use the git version of inkex.

Hope this is a little bit of help.

**Comment by bigredfrog** [comment]
Thanks for the response, will follow up and try these steps later today. I think I have inkex removed already.

Biggest concern from my ramblings above is that I had to remove yarn.lock

This could be a very bad thing, as I understand it is effectively a lock of versions to use in electron build. Should I go back and try and make that work without, or ignore for now?

Also note that with the git clone of head I believe I see a nasty regression on pes output compared to v2.0.0

I have a layer which is turned off in the objects stack as it is only assistance for my design flow being exported as stitch output...

**Comment by kaalleen** [comment]
I am a bit short on time, so here a quick comment...

I don't think that removing the yarn.lock is a big concern. The versions are specified in the package.json file, so you will still end up with the correct versions. I am not sure though, why it had been necessary. Never had that issue.

> Also note that with the git clone of head I believe I see a nasty regression on pes output compared to v2.0.0

In which way? What is the difference? I am not sure if I understand.

> I have a layer which is turned off in the objects stack as it is only assistance for my design flow being exported as stitch output...

Does it also happen, when you download a recent release, such as this one here: https://github.com/inkstitch/inkstitch/releases/tag/dev-build-kaalleen-update-nodejs

There had been a problem with adding layer commands - as you pointed out in a previous comment. It's a bug and needs to be fixed. Thank you for this one.

**Comment by bigredfrog** [comment]
Thanks I appreciate any help, and also recognise real life is a thing!

Good to know of the yarn build.

I will move to the branch you have flagged, off the v2.0.0 and see if the error expresses, I will add images and explanation under a new ticket if it does.

**Comment by bigredfrog** [comment]
I might need some pointers on how to integrate these nodejs changes, I updated my branches to new from remote, and I believe I have moved to it with checkout, but trying to work out the correct steps to be running this nodejs or is it just automagical?

Will try institch plugin shortly anyway

**Comment by bigredfrog** [comment]
Yep, I think I need some education here.

I have dug around in the commits in your branch, and I can see it changes git behaviours in submissions and that it removes "fibers"

However, if I am understanding things correctly I should see a response from running 

node-v 

of v12.21.0 from the command line. Assuming that is what inkstitch will run? Wondering how I can tell, I will have in a command of such as an exit('') in a simple menu call to confirm.

Assuming I should be seeing 16.x, but I am a stumbling noob at this point.

**Comment by bigredfrog** [comment]
Still pelunking, looking at getting debbuger attached to pycharm, not something I am experienced with, but has to be better than inkex.utils.debug() which exits on output!

Meanwhile, trying to save a .pes on this branch fails with

Ink/Stitch experienced an unexpected error.



If you'd like to help, please file an issue at https://github.com/inkstitch/inkstitch/issues and include the entire error description below:



Ink/Stitch Version: Manual Install (Linux)





Traceback (most recent call last):

  File "/home/big_red_frog/dev/inkstitch/inkstitch.py", line 66, in <module>

    extension.run(args=remaining_args)

  File "/usr/share/inkscape/extensions/inkex/base.py", line 131, in run

    self.save_raw(self.effect())

  File "/home/big_red_frog/dev/inkstitch/lib/extensions/output.py", line 50, in effect

    if not self.get_elements():

  File "/home/big_red_frog/dev/inkstitch/lib/extensions/base.py", line 175, in get_elements

    self.elements = nodes_to_elements(self.get_nodes(troubleshoot))

  File "/home/big_red_frog/dev/inkstitch/lib/extensions/base.py", line 172, in get_nodes

    return self.descendants(self.document.getroot(), troubleshoot=troubleshoot)

  File "/home/big_red_frog/dev/inkstitch/lib/extensions/base.py", line 159, in descendants

    nodes.extend(self.descendants(child, selected, troubleshoot))

  File "/home/big_red_frog/dev/inkstitch/lib/extensions/base.py", line 141, in descendants

    element.get_style('display', 'inline') is None):

  File "/home/big_red_frog/dev/inkstitch/lib/elements/element.py", line 171, in get_style

    style = self._get_specified_style().get(style_name, default)

  File "/home/big_red_frog/dev/inkstitch/lib/elements/element.py", line 168, in _get_specified_style

    return self.node.specified_style()

  File "/usr/share/inkscape/extensions/inkex/elements/_base.py", line 165, in __getattr__

    raise AttributeError(f"Can't find attribute {self.typename}.{name}")

AttributeError: Can't find attribute Layer.specified_style

**Comment by bigredfrog** [comment]
Well thats annoying, looks like the bits of pycharm I need for remote debugging are only fully activated in the professional version :-( Though I may be getting things wrong.

**Comment by bigredfrog** [comment]
So I switched over to eclipse so I can run pydevd without having to upgrade pycharm
Used the inbuilt git interfacing to pull everything, rebuilt stuff, and deleted the old code branches so I know I am running clean.
I can get the debugger to breakpoint into inkstitch.py, but not the supporting libs yet, so that is progress.

Actually got it to break point in files like color.py now, by right click in project explorer in lib and pydev / set as source folder

So now I have full debug attachement to the running inkscape extensions which will help greatly.

I also played with hacking in some non terminal logging lib and that works as well, so best of both worlds there.

As I am still spelunking I would appreciate some advice, do you want me to keep ramblings in here until I have a successful baseline?

Or open new issues.

I am presently on main  head, and I will try and flip over now to dev-build-kaalleen-update-node-js once I have my head around the git intergration into eclipse.

Well if you measure progress by learning from mistakes, right now, I am going sub orbital!

**Comment by bigredfrog** [comment]
So I am pretty sure I am on the right branch

![image](https://user-images.githubusercontent.com/7328740/127898423-78648382-0ad3-4717-8be7-65b0a35bc3da.png)

Its a branch I have pulled from your nodejs updates

and I do still see the extra layers in the .PES

None of these should of been in the export, as they were all turend off ( eye closed ) in the .svg when I did a save as .PES 

![image](https://user-images.githubusercontent.com/7328740/127898264-2179de2c-4148-4d33-ae2e-14779efaa85e.png)

I will see if flipping back to the release makes this go away...

**Comment by bigredfrog** [comment]
Now I am feeling dumb

I can clearly see there is a v2.0.0 tag release

https://github.com/inkstitch/inkstitch/releases/tag/v2.0.0

But I cannot see it as a tag I can sync to in git via eclipse, I see just about everything else such as under "v1"

But nothing for v2

![image](https://user-images.githubusercontent.com/7328740/127906523-95d0578a-2978-4eef-b97b-2f89a3a01694.png)

I think there is something I don't understand about that v2.0.0 tag in github :-(

**Comment by bigredfrog** [comment]
Somehow got eclipse to pull ALL tags, so now I have v2.0.0 and some others I could not find.

So I can comfortably switch between v2.0.0 and https://github.com/inkstitch/inkstitch/releases/tag/dev-build-kaalleen-update-nodejs

Done a double swap to ensure I am not missing something, basically v2.0.0 runs clean for my export of .pes with only the expected layers.

The nodejs branch has the extra layers exported as described above, and from just pulling from head.

Happy to share the SVG if it helps ( own original artwork etc )

My workflow when changing branch is, check out the tag I want.

edit requirements.txt again to comment out pyemroidery
pip install -r requirements.txt
cd electron
yarn install
cd ..
make inx
delete extra languages in inx folder
Check I have not ended up with a pip install inkex with 
pip show inkex

So I will continue to scratch off the v2.0.0 tag for now

I do still have that fibers problem when trying to launch the simulation, so I may try and steal across those tweaks from electron.
Same issue also stops import of thread color pallets.

Params and preview work nicely.

So next step then for me is to see if I can hand port over the changes to remove fibers as per your branch and see what happens.

I am definately making progress on my spelunking, and almost have somewhere to stand...

**Comment by bigredfrog** [comment]
Removing the fibers from package.json in electron stopped that particular complaint, but then I hit a problem where I think the debugger that I am running for python clashes with the debugger for nodejs.

I turn off the debugger entirely by removing the file "DEBUG" and then I just get hwat I think is a generic warning and now actual simulation

So I will try and avoid workflows that depend on electron right now :-(

/home/big_red_frog/git/inkstitch/lib/api/server.py:56: UserWarning: The 'environ['werkzeug.server.shutdown']' function is deprecated and will be removed in Werkzeug 2.1.

  request.environ.get('werkzeug.server.shutdown')()

As always, any tips appreciated

**Comment by kaalleen** [comment]
The werkzeug shutdown message is just a warning. It doesn't stop electron from working. In the main branch it should be fixed though.

That's right, simulator, PDF export and installing the color palettes rely on electron.

**Comment by bigredfrog** [comment]
Your fix on add layers helped a lot for my experience on kaalleen/update-nodejs

But none of my electron workflows are successful, just nothing gets launched, I get the spinner, and then that dialog which I appreciate is just showing anything that was reported on completion. I know what to expect from a non development install, but something must be wrong with how my dev environment is set up, if you state the electron should work.

So close now :-(

**Comment by bigredfrog** [comment]
HA, well I don't know if this is hilarious in reality, or lack of understanding on my part.

So I moved to head of main, as I could see changes propogating in for fibers, add_layers etc.

Things were mostly working, ( unwanted layer export aside ) except for anything in electron, and I could not work out why.

The error was that the port required to run the simulator etc was 8090. 

So I went hunting in htop and started killing off node instances as I suspected maybe I orphaned some processes in earlier developement, but this closed eclipse which I am using for python remote debugging, and the electron parts just work!

Eclipse must use nodejs on port 8090. I don't yet know how easy it is to change that in eclipse or in the electron launched from the inkstitch extension. Nor have I double blind validated this hypothesis.

Simulator output runs, as commented though with signficant number of layers that should not be active, same as .pes export

Same for pdf export, shows the extra layers but otherwise runs, and I can run the interface for pallet import.
